### DISCUSSION TITLE
Reintegration of Poof/Ube fork into main (celo-tools)

### DISCUSSION THREAD

**Discussion by aaronmgdr** [celo-org/react-celo#125] [General]
Currently their are two primary versions of `use-contractkit`, the celo-tools version maintained by cLabs Applications team, and the [poofcash/ubeswap/nomspace version](https://github.com/Ubeswap/use-contractkit/tree/master/packages/use-contractkit) forked by @BriungRi and used by many of the dapps in celo ecosystem such as moola. 

This the place to sort out which of the divergent set of features will be needed to be supported by celo-tools version for the [reunion of these codebases.](https://github.com/celo-org/use-contractkit/compare/master...Ubeswap:master?expand=1) 

The main difference is that the poof fork relies on dappkit to connect to valora whereas the celo-tools version use wallet connect. Lets talk about what that enables.  

What functionality is needed to be brought in

**Comment by aaronmgdr** [comment]
Poof Version will redirect Back to browser after use approves tx in valora

**Reply by aaronmgdr** [reply]
This is because of DappKit. 

@jeanregisser  / valora would need to implement in how valora supports wallet connct. 

Note that while loved, this feature also has its bugs.

**Reply by aaronmgdr** [reply]
@jeanregisser 

do you think valora will add in the feature to mimic dappkit's inter app navigation when using wallet-connect? (ie return to previous app). 

One tricky part might be handling desktop to mobile (where we dont want navigating) vs mobile app to mobile app (where we do)

**Reply by aaronmgdr** [reply]
Given that Valora now supports in app browsing to dapps we (the clabs apps team) have decided this is a wont fix. that is we will not be moving back to dapp kit and instead use wallet connect

**Comment by aaronmgdr** [comment]
Poof Version supports other chains

**Reply by aaronmgdr** [reply]
I think by default we should only support celo networks. However making it easy to use contract kit for cross chain apps seems like a good idea. For best support I would say we should hide celo only wallets like Valora if on a network they dont support. to this end I would add a field to providers on which chain ids they support.

**Reply by aaronmgdr** [reply]
react-celo now supports completely arbitrary networks array, pass in any evm chain id, name, rpc url you desire

**Comment by aaronmgdr** [comment]
Poof Version supports some wallet that celo-tools version does not like iMToken

**Comment by aaronmgdr** [comment]
Poof Version behavior around metamask on mobile browser is different

**Reply by aaronmgdr** [reply]
This has been addressed. both now open metamask mobile app

**Comment by dckesler** [comment]
The dappkit vs walletconnect is probably the reason why the poof version doesn't have the timeouts people are seeing. That's a wallet connect issue.

**Comment by jeanregisser** [comment]
Poof's version still uses WalletConnect v2, which makes it impossible to use it with Valora for the desktop <-> mobile use case.
I'm looking forward for this to be switched back to WC v1.

**Reply by jmrossy** [reply]
Yep, ditto for Celo Wallet. Ubeswap doesn't work with Celo Wallet for this reason

**Reply by dckesler** [reply]
I think our idea with re-integrating the fork would be to stay on WCv1 but include dappkit support. Depending on the scope of changes to the fork we may even just want to add back in dappkit support ourselves.

**Comment by dckesler** [comment]
Seems like this would also address #129
